Tree/AVL_Trees/AVL_Tree.c

```diff
@@ -0,0 +1,427 @@
+        return root;
+    }
+
+    else if(data < root->info)
+        root->left = AVLinsertion(root->left , data);
+
+    else
+        root->right = AVLinsertion(root->right , data);
+
+    //node's height will be max of (l-height,r-height) + 1
+    root->height = (max(height(root->left) , height(root->right))+1);
+
+    //Insertion is done now conditions for rotation if needed.
+    if(BalanceFactor(root)==2 && BalanceFactor(root->left)==1)
+        return LLrotation(root);
+
+    if(BalanceFactor(root)==2 && BalanceFactor(root->left)==-1)
+        return LRrotation(root);
+
+    if(BalanceFactor(root)==-2 && BalanceFactor(root->right)==-1)
+        return RRrotation(root);
+
+    if(BalanceFactor(root)==-2 && BalanceFactor(root->right)==1)
+        return RLrotation(root);
+
+    return root;
+}
+
+struct node* minimumnode(struct node *root)
+{
+    if(root == NULL)
+        return NULL;
+
+    if(root->left == NULL)
+        return root;
+
+    else
+        minimumnode(root->left);
+}
+
+struct node* maximumnode(struct node *root)
+{
+    if(root == NULL)
+        return NULL;
+
+    if(root->right == NULL)
+        return root;
+
+    else
+        maximumnode(root->right);
+}
+
+struct node* deleteelement(struct node *root , int data)
+{
+
+    if(root == NULL)
+        return root;
+
+    if(data < root->info)
+        root->left = deleteelement(root->left , data);
+
+    else if(data > root->info)
+        root->right = deleteelement(root->right , data);
+
+    else
+    {
+        if(root->left != NULL && root->right != NULL)
+        {
+            struct node* temp = minimumnode(root->right);
+            root->info = temp->info;
+            root->right = deleteelement(root->right , temp->info);
+        }
+
+        else if(root->left != NULL)
+            return (root->left);
+
+        else if(root->right != NULL)
+            return(root->right);
+
+        else                                                                     //both left and left is null
+            return NULL;
+    }
+
+    if(root == NULL)
+        return root;
+
+    root->height = 1+(max(height(root->left) , height(root->right)));
+
+    if(BalanceFactor(root)==2 && BalanceFactor(root->left)==1)
+        return LLrotation(root);
+
+    else if(BalanceFactor(root)==2 && BalanceFactor(root->left)==-1)
+        return LRrotation(root);
+
+    else if(BalanceFactor(root)==-2 && BalanceFactor(root->right)==-1)
+        return RRrotation(root);
+
+    else if(BalanceFactor(root)==-2 && BalanceFactor(root->right)==1)
+        return RLrotation(root);
+
+    return root;
+}
+
+int checkAVL(struct node* root)
+{
+    if(root == NULL)
+        return 1;
+
+    int left = checkAVL(root->left);
+    int right = checkAVL(root->right);
+
+    if(BalanceFactor(root) >= -1 && BalanceFactor(root) <= 1)
+        return 1;
+
+    return 0;
+}
+
+
+
+int main()
+{
+    struct node *root = NULL;
+    int data,choice;
+    while(1)
+    {
+        printf("\n1. Insertion");
```

Comment:
You can do something here, or the function can be modified

Reply:
I actually added a message to input integer only. Will it be okay?